### PR TITLE
Different dtypes testing

### PR DESCRIPTION
**Pull request by TheSlimvReal**
## Description

After I used the TestSuite for the first time in production I found some useful functionalities to extend the module with.

As @coquelin77 requested, the TestSuite is now testing on multiple dtypes which can also be defined by the user.
Internally it used `np.random.randn` for float values and `np.random.randint` for integer values.
The latter one also accepts arguments for upper and lower bound of the random values.

Added to that, debugging messages were made more precise about when a exception is risen.

## Type of change

Select relevant options.

[ ] Bug fix (non-breaking change which fixes an issue)
[x] New feature (non-breaking change which adds functionality)
[ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
[ ] Documentation update

Are all split configurations tested and accounted for?
[x] yes [ ] no
Does this change require a documentation update outside of the changes proposed?
[ ] yes [x] no
Does this change modify the behaviour of other functions?
[ ] yes [x] no
Are there code practices which require justification?
[ ] yes [x] no

### CODE REVIEW

File: heat/core/tests/test_suites/basic_test.py [hunk 1]
```diff
@@ -118,36 +128,111 @@ def assert_array_equal(self, heat_array, expected_array):
...
+        self,
+        tensor,
+        heat_func,
+        numpy_func,
+        heat_args=None,
+        numpy_args=None,
+        distributed_result=True,
+    ):
+        """
+        This function will create random tensors of the given shape with different data types.
```
Comment (coquelin77) [inline heat/core/tests/test_suites/basic_test.py]: these docs are the same as for assert_func_equal. are these functions the same?

File: heat/core/tests/test_suites/basic_test.py [hunk 2]
```diff
@@ -209,9 +293,50 @@ def assert_func_equal(
             else:
                 self.assertTrue(np.array_equal(ht_res._DNDarray__array.numpy(), np_res))
 
-    def _create_random_array(self, shape):
+    def _create_random_array(self, shape, dtype=np.float64, low=-10000, high=10000):
```
Comment (coquelin77) [inline heat/core/tests/test_suites/basic_test.py]: can you note that this creates a numpy array? others might misjudge it by its name, maybe just change array to np_array? just a thought

**Comment by coquelin77** [comment]
also can you trigger a new coverage run?

**Comment by TheSlimvReal** [comment]
I removed the communicator tests in favour of #422. Once this discussion comes to a conclusion, the comm test might be added again.

File: heat/core/tests/test_suites/basic_test.py (hunk 1 above)
Comment (TheSlimvReal) [reply heat/core/tests/test_suites/basic_test.py]: That was a mistake, I updated the docstring

File: heat/core/tests/test_suites/basic_test.py (hunk 2 above)
Comment (TheSlimvReal) [reply heat/core/tests/test_suites/basic_test.py]: I changed the name to `__create_random_np_array`

**Pull request merged**